extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java

```diff
@@ -30,15 +38,65 @@
 @ApplicationScoped
 // TODO: create a different entry for WorkerPoolRegistry than `analyzeWorker` and drop this class
 public class QuarkusWorkerPoolRegistry extends WorkerPoolRegistry {
+
+    static final Logger logger = Logger.getLogger("io.quarkus");
     private static final String WORKER_CONFIG_PREFIX = "smallrye.messaging.worker";
     private static final String WORKER_CONCURRENCY = "max-concurrency";
+    public static final String DEFAULT_VIRTUAL_THREAD_WORKER = "<virtual-thread>";
+    public static final int DEFAULT_VIRTUAL_THREAD_MAX_CONCURRENCY = 5000;
 
     @Inject
     ExecutionHolder executionHolder;
 
+    private final Map<String, Semaphore> virtualThreadConcurrency = new ConcurrentHashMap<>();
```

Comment:
Do we need a map here?  or for virtual threads we do have a single (named) pool?
If it's the case you can just use a volatile `ConcurrencyLimiter`'s record intance which would contains the `pool name` and the `Semaphore`

Reply:
for now there's only one virtual executor but we might have more of them in the future ? I could imagine a scenario where you could specify @RunOnVirtualThread(strategy="fifo") on the model of kotlin coroutines where several coroutines dispatchers exist ?

I don't know if we should leave it as-is for now or change it to something lighter

Reply:
When we would end up having more I suppose  they would be brought per Netty event loop (or something like that), but I am sure this class would likely change the same, given that we would have the precise enumeration and identity of them

Reply:
That has been removed.